solr/modules/aws-secret-provider/src/java/org/apache/solr/secret/zk/AWSSecretManagerCredentialsInjector.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.secret.zk;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.lang.invoke.MethodHandles;
+import java.util.Collections;
+import java.util.List;
+import org.apache.solr.common.cloud.ZkCredentialsInjector;
+import org.apache.solr.common.util.StrUtils;
+import org.apache.solr.util.SolrJacksonAnnotationInspector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClientBuilder;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.SecretsManagerException;
+
+public class AWSSecretManagerCredentialsInjector implements ZkCredentialsInjector {
+  public static final String SECRET_CREDENTIAL_PROVIDER_SECRET_NAME_VM_PARAM =
```

Comment:
Actually, they are not consistent because they don't have the same functionality. Some are VM variable names (internal to Solr, so I followed the same naming convention of the existing creds/acl providers), one is an AWS variable value. 
How about this new version?


  ```
   public static final String AWS_SM_CREDENTIALS_SECRET_NAME_VM_PARAM = "zkAWSSecretCredentialsSecretName";
  private static final String AWS_SM_CREDENTIALS_REGION_VM_PARAM = "zkCredentialsAWSSecretRegion";
  private static final String AWS_SM_CREDENTIALS_SECRET_NAME_DEFAULT = "/solr/zk/credentials/dev/secret";
```


Followed AWS SM [recommended](https://aws.amazon.com/blogs/security/migrating-your-secrets-to-aws-secrets-manager-part-i-discovery-and-design/) naming convention.